deltacat/compute/compactor_v2/utils/primary_key_index.py

```diff
@@ -167,6 +167,40 @@ def group_by_pk_hash_bucket(
     return result
 
 
+def _sliced_string_cast(array: pa.ChunkedArray) -> pa.ChunkedArray:
```

Comment:
would moving it to pyarrow utils and take in second arg as type make it more generic?

Reply:
moved over to the utils. What would the second arg express?

Reply:
The type to cast it to i.e., `pa.string()`?

Reply:
this function is only relevant for casting to `string` types. So would the second arg be either `pa.string()` and `pa.large_string()`?

Or are you thinking to make this a generic casting function for all types, special casing when we see `pa.string()`?

Reply:
Yeah, I was thinking this sliced cast is relevant for all target types, but not just strings. We could only support strings for now.